packages/rum-angular/src/apm.service.ts

```diff
@@ -24,19 +24,25 @@
  */
 
 import { Router, NavigationStart } from '@angular/router'
-import { Injectable } from '@angular/core'
+import { Inject, Injectable, NgZone } from '@angular/core'
 import { afterFrame } from '@elastic/apm-rum-core'
+import { ApmBase } from '@elastic/apm-rum'
+import { APM } from './apm.module'
 
 @Injectable({
   providedIn: 'root'
 })
 export class ApmService {
-  static apm: any
-
-  constructor(public router: Router) {}
+  constructor(
+    @Inject(APM) public apm: ApmBase,
+    private router: Router,
+    private readonly ngZone: NgZone
+  ) {}
 
   init(config) {
-    const apmInstance = ApmService.apm.init(config)
+    const apmInstance = this.ngZone.runOutsideAngular(() =>
+      this.apm.init(config)
```

Comment:
Why does the init has to be out side the zone?

Reply:
Check the discussion here -  https://github.com/elastic/apm-agent-rum-js/pull/971#discussion_r583328989